sdk/common/src/main/java/io/opentelemetry/sdk/common/export/MemoryMode.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.common.export;
+
+/** The type of memory allocation used during signal collection in the different readers. */
```

Comment:
Maybe something like? No point making it more specific until its adopted in the different providers.
```suggestion
/** The memory semantics of the SDK. */
```